### PR TITLE
Schema refinements

### PR DESCRIPTION
**Pull request by mwalzer**
4+ eyes principle: @bittremieux , can you have a look if I broke something before merge please?

### CODE REVIEW

File: schema/v0_1_0/mzqc_0_1_0.schema.json [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+				"setQualities": {
+					"description": "List of setQuality elements.",
+					"type": "array",
+					"minItems": 1,
+					"items": {
+						"$ref": "#/definitions/setQuality"
+					}
+				},
+				"controlledVocabularies": {
+					"description": "Collection of CV elements.",
```
Comment (bittremieux) [inline schema/v0_1_0/mzqc_0_1_0.schema.json]: Rename CV -> controlledVocabulary.

File: schema/v0_1_0/mzqc_0_1_0.schema.json [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+		"setQuality": {
+			"description": "Parent element describing combined QC metrics for multiple runs.",
+			"$ref": "#/definitions/baseQuality"
+		},
+		"cvParameter": {
+			"description": "Base element containing a reference to a controlled vocabulary.",
+			"type": "object",
+			"properties": {
+				"cvRef": {
+					"description": "Reference to the CV that contains the parameter definition.",
```
Comment (bittremieux) [inline schema/v0_1_0/mzqc_0_1_0.schema.json]: Rename CV -> controlledVocabulary.

File: schema/v0_1_0/mzqc_0_1_0.schema.json [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+					"type": "string",
+					"pattern": "^[A-Z]+$"
+				},
+				"accession": {
+					"description": "Accession number identifying the parameter within its CV.",
```
Comment (bittremieux) [inline schema/v0_1_0/mzqc_0_1_0.schema.json]: Spell out CV -> controlled vocabulary.

File: schema/v0_1_0/mzqc_0_1_0.schema.json [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+					"type": "string",
+					"pattern": "^[A-Z]+:[0-9]{7}$"
+				},
+				"name": {
+					"description": "Name of the CV element describing the parameter.",
```
Comment (bittremieux) [inline schema/v0_1_0/mzqc_0_1_0.schema.json]: Spell out CV -> controlled vocabulary.

File: schema/v0_1_0/mzqc_0_1_0.schema.json [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+				},
+				"description": {
+					"description": "Description of the parameter.",
+					"type": "string"
+				},
+				"value": {
+					"description": "Value of the parameter."
+				},
+				"unit": {
+					"description": "A CV element describing the unit of the parameter value.",
```
Comment (bittremieux) [inline schema/v0_1_0/mzqc_0_1_0.schema.json]: Rename CV -> cvParameter.

File: schema/v0_1_0/mzqc_0_1_0.schema.json [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+				}
+			},
+			"required": ["cvRef", "accession", "name"]
+		},
+		"metadata": {
+			"description": "Metadata describing the QC analysis.",
+			"type": "object",
+			"properties": {
+				"fileProvenance": {
+					"description": "Optional string describing/documenting the provenance of the file",
```
Comment (bittremieux) [inline schema/v0_1_0/mzqc_0_1_0.schema.json]: This is pretty vague. Whereas proper file provenance is very important, the way this property is right know I don't know how useful it'd be.

File: schema/v0_1_0/mzqc_0_1_0.schema.json [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+		"qualityMetric": {
+			"description": "Parameter describing a QC metric.",
+			"$ref": "#/definitions/cvParameter"
+		},
+		"controlledVocabulary": {
+			"description": "Element describing a controlled vocabulary.",
+			"type": "object",
+			"properties": {
+				"name": {
+					"description": "Full name of the CV.",
```
Comment (bittremieux) [inline schema/v0_1_0/mzqc_0_1_0.schema.json]: Spell out CV -> controlled vocabulary.

File: schema/v0_1_0/mzqc_0_1_0.schema.json [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+					"type": "string"
+				},
+				"ref": {
+					"description": "Reference string to denote the use of this specific CV by.",
```
Comment (bittremieux) [inline schema/v0_1_0/mzqc_0_1_0.schema.json]: Spell out CV -> controlled vocabulary.

File: schema/v0_1_0/mzqc_0_1_0.schema.json [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+					"type": "string"
+				},
+				"uri": {
+					"description": "Publicly accessible URI of the CV.",
```
Comment (bittremieux) [inline schema/v0_1_0/mzqc_0_1_0.schema.json]: Spell out CV -> controlled vocabulary.

File: schema/v0_1_0/mzqc_0_1_0.schema.json [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+					"type": "string",
+					"format": "uri"
+				},
+				"version": {
+					"description": "Version of the CV.",
```
Comment (bittremieux) [inline schema/v0_1_0/mzqc_0_1_0.schema.json]: Spell out CV -> controlled vocabulary.

File: schema/v0_1_0/mzqc_0_1_0.schema.json (hunk 1 above)
Comment (mwalzer) [reply schema/v0_1_0/mzqc_0_1_0.schema.json]: :+1:

File: schema/v0_1_0/mzqc_0_1_0.schema.json (hunk 6 above)
Comment (mwalzer) [reply schema/v0_1_0/mzqc_0_1_0.schema.json]: I agree that it is important for a number of use cases. So, it was optional. Would you rather abolish it completely or reshape/refine it.

Comment (mwalzer) [reply schema/v0_1_0/mzqc_0_1_0.schema.json]: We had a discussion during the call on that. It seems this provenance can be fairly complex, so we better take the decision to not include provenance even as optional in the schema, rather leave it for a later revision of the format with a focus on this particular feature. Like that, we will not have conflicting 'objects of provenance'.

**Pull request merged**